packages/data-table-server/src/dataTableService/userDefined/SqlDataTableService.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { yup } from '@tupaia/utils';
+import { DataTableServerModelRegistry } from '../../types';
+import { UserDefinedDataTableService } from './UserDefinedDataTableService';
+
+const configSchema = yup.object().shape({
+  externalDatabaseConnectionCode: yup.string().required(),
```

Comment:
Chose to go with `code` over `id`. Arguably `id` is better since it's more stable, but I doubt we're going to change these codes often/ever and `code` is a lot more readable 👍